source_gen/lib/builder.dart

```diff
@@ -131,9 +137,12 @@ class CombiningBuilder implements Builder {
     final ignoreForFile = _ignoreForFile.isEmpty
         ? ''
         : '\n// ignore_for_file: ${_ignoreForFile.join(', ')}\n';
+
+    final preamble = _preamble.isEmpty ? '' : '\n$_preamble\n';
+
     final output = '''
 $defaultFileHeader
-${languageOverrideForLibrary(inputLibrary)}$ignoreForFile
+${languageOverrideForLibrary(inputLibrary)}$ignoreForFile$preamble
```

Comment:
There is some question as to exactly where this should go, but I think probably after the ignore for file lines is fine?

We should mention in the docs that this is where it will go though

Reply:
You could argue for `preamble` as the more general option beeing first and `ignoreForFile` as the more specific beeing second. But its a matter of taste i guess

Reply:
yeah I don't think it matters much either way? we should just document which is which :)

Reply:
Done